### ISSUE TITLE
Move unit test code from contract to most specific module

### ISSUE THREAD

**Issue by sesuncedu** [owlcs/owlapi#591]
Test code in contract that is really testing code in a specific module should move to that module.

**Comment by ignazio1977** [comment]
Agreed when it's easy to do. For some tests, this might require API changes; a common situation is when an OWLOntologyManager is required.

**Comment by sesuncedu** [comment]
I've been doing this a few at a time. I need to merge the changes onto a clean version5, as I ended up with some drift.
